Makefile

```diff
@@ -13,17 +18,17 @@ LIBGIT2_LIB_PATH := $(LIBGIT2_PATH)/lib
 LIBGIT2_LIB64_PATH := $(LIBGIT2_PATH)/lib64
 LIBGIT2 := $(LIBGIT2_LIB_PATH)/libgit2.a
 MUSL-CC =
+LIBGIT2_ONLY ?=
 
 export CGO_ENABLED=1
 export LIBRARY_PATH=$(LIBGIT2_LIB_PATH):$(LIBGIT2_LIB64_PATH)
 export PKG_CONFIG_PATH=$(LIBGIT2_LIB_PATH)/pkgconfig:$(LIBGIT2_LIB64_PATH)/pkgconfig
 export CGO_CFLAGS=-I$(LIBGIT2_PATH)/include
 
-
 ifeq ($(shell uname -s),Linux)
 	export CGO_LDFLAGS=$(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --libs --static --cflags libssh2 openssl libgit2) -static
 else
-	export CGO_LDFLAGS=$(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --libs --static --cflags libssh2 openssl libgit2) -Wl,--unresolved-symbols=ignore-in-object-files -Wl,-allow-shlib-undefined -static
+	export CGO_LDFLAGS=$(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config --libs --static --cflags libgit2)
```

Comment:
Does this change work in Darwin? I am assuming the target of this change is `LIBGIT2_ONLY` and not `!Linux`.

Reply:
Yep it works, I haven't checked on Darwin amd64 though.
This led me to realize that we check if everything works on macOS by running the build script [here](https://github.com/fluxcd/golang-with-libgit2/blob/main/.github/workflows/build.yaml#L20-L40) but this is redundant since we anyway run the build scripts in the workflow that builds and publishes the libs. We should be running `make dev-test` here instead to make sure that tests pass on macOS (we run `make test` which runs the tests in a container before publishing the docker images, so I assume it's fine to skip Linux)

Reply:
Nice catch, go ahead changing it with `make dev-test`.